prover/src/lib.rs

```diff
@@ -257,15 +253,12 @@ pub trait Prover {
         // 1 ----- Commit to the execution trace --------------------------------------------------
 
         // build computation domain; this is used later for polynomial evaluations
-        #[cfg(feature = "std")]
-        let now = Instant::now();
-        let domain = StarkDomain::new(&air);
-        #[cfg(feature = "std")]
-        debug!(
-            "Built domain of 2^{} elements in {} ms",
-            domain.lde_domain_size().ilog2(),
-            now.elapsed().as_millis()
-        );
+        let lde_domain_size = air.lde_domain_size();
+        let domain = {
+            let _ = info_span!("build_domain", domain_size = lde_domain_size).entered();
+            StarkDomain::new(&air)
+        };
+        assert_eq!(domain.lde_domain_size(), lde_domain_size);
 
         // extend the main execution trace and build a Merkle tree from the extended trace
         let (mut trace_lde, mut trace_polys): (Self::TraceLde<E>, TracePolyTable<E>) =
```

Comment:
Let's create a span for this step as well. This way, we'd have one high-level span with more detailed spans under it. It could look something like this:
```Rust
// commit to the main trace segment
let (mut trace_lde, mut trace_polys) = {
    // extend the main execution trace and build a Merkle tree from the extended trace
    let span = info_span!("commit_to_main_trace_segment").entered();
    let (trace_lde, trace_polys) =
        self.new_trace_lde(&trace.get_info(), trace.main_segment(), &domain);

    // get the commitment to the main trace segment LDE
    let main_trace_root = trace_lde.get_main_trace_commitment();

    // commit to the LDE of the main trace by writing the root of its Merkle tree into
    // the channel
    channel.commit_trace(main_trace_root);

    drop(span);
    (trace_lde, trace_polys)
};
```